sklearn/cluster/tests/test_k_means.py

```diff
@@ -349,20 +424,24 @@ def test_k_means_fit_predict(algo, dtype, constructor, seed, max_iter, tol):
         pytest.xfail(
             "Known failures on MacOS, See "
             "https://github.com/scikit-learn/scikit-learn/issues/12644")
-    if not (algo == 'elkan' and constructor is sp.csr_matrix):
-        rng = np.random.RandomState(seed)
 
-        X = make_blobs(n_samples=1000, n_features=10, centers=10,
-                       random_state=rng)[0].astype(dtype, copy=False)
-        X = constructor(X)
+    rng = np.random.RandomState(seed)
```

Comment:
Is the note above about small chances of failure with elkan still True?

Reply:
I'm not entirely sure. The reason of the failure still holds so for safety I'd rather just leave it for now.

Reply:
I'm asking because you have removed the condition that skips the test otherwise

Reply:
(It's mostly me trying to understand the changes, I have no opinion whether this is good or bad. If tests pass, that's fine)

Reply:
I removed the skip when `elkan + sparse` because now elkan supports sparse. But I kept the xfail part.